crates/sui-indexer/src/apis/read_api_v2.rs

```diff
@@ -18,19 +20,46 @@ use sui_json_rpc_types::{
     SuiTransactionBlockResponse, SuiTransactionBlockResponseOptions,
 };
 use sui_open_rpc::Module;
+use sui_protocol_config::{ProtocolConfig, ProtocolVersion};
 use sui_types::base_types::{ObjectID, SequenceNumber};
-use sui_types::digests::TransactionDigest;
+use sui_types::digests::{ChainIdentifier, TransactionDigest};
 use sui_types::sui_serde::BigInt;
 
 use sui_json_rpc_types::SuiLoadedChildObjectsResponse;
 
+#[derive(Clone)]
 pub(crate) struct ReadApiV2 {
-    pg_store: PgIndexerStoreV2,
+    inner: IndexerReader,
 }
 
 impl ReadApiV2 {
-    pub fn new(pg_store: PgIndexerStoreV2) -> Self {
-        Self { pg_store }
+    pub fn new(inner: IndexerReader) -> Self {
+        Self { inner }
+    }
+
+    async fn get_checkpoint(&self, id: CheckpointId) -> Result<Checkpoint, IndexerError> {
+        match self
+            .inner
+            .spawn_blocking(move |this| this.get_checkpoint(id))
+            .await
+        {
+            Ok(Some(epoch_info)) => Ok(epoch_info),
+            Ok(None) => Err(IndexerError::InvalidArgumentError(format!(
+                "Missing checkpoint {id:?}"
```

Comment:
nit: "missing" sounds problematic, while it could be an user error. Should we say "checkpoint {id:?} not found"?